bootstrap/cli.py

```diff
@@ -0,0 +1,141 @@
+    do_build(minifi_options)
+    return True
+
+
+def modify_bool_options(minifi_options: MinifiOptions):
+    options = [inquirer.Checkbox(
+        "MiNiFi C++ options",
+        message="Select MiNiFi C++ components",
+        choices=[name for name, obj in minifi_options.bool_options.items()],
+        default=[name for name, obj in minifi_options.bool_options.items() if obj.value == "ON"]
+    )]
+
+    selection_result = inquirer.prompt(options)
+    for minifi_option in minifi_options.bool_options.values():
+        if minifi_option.name in selection_result:
+            minifi_option.value = "ON"
+        else:
+            minifi_option.value = "OFF"
+
+
+def main_menu(minifi_options: MinifiOptions):
+    done = False
+    while not done:
+        main_menu_options = {
+            f"Build dir: {minifi_options.build_dir}": build_dir_menu,
+            f"Build type: {minifi_options.build_type.value}": build_type_menu,
+            "Build options": bool_menu,
+            "One click build": do_one_click_build,
+            "Step by step build": step_by_step_menu,
+            "Exit": lambda options: True,
+        }
+
+        questions = [
+            inquirer.List(
+                "sub_menu",
+                message="Main Menu",
+                choices=[menu_option_name for menu_option_name in main_menu_options],
+            ),
+        ]
+
+        main_menu_prompt = inquirer.prompt(questions)
+        done = main_menu_options[main_menu_prompt["sub_menu"]](minifi_options)
+
+
+def build_type_menu(minifi_options: MinifiOptions) -> bool:
+    questions = [
+        inquirer.List(
+            "build_type",
+            message="Build type",
+            choices=minifi_options.build_type.possible_values,
+        ),
+    ]
+
+    answers = inquirer.prompt(questions)
+    minifi_options.build_type.value = answers["build_type"]
+    return False
+
+
+def build_dir_menu(minifi_options: MinifiOptions) -> bool:
+    questions = [
+        inquirer.Path('build_dir',
+                      message="Build directory",
+                      default=minifi_options.build_dir
+                      ),
+    ]
+    minifi_options.build_dir = inquirer.prompt(questions)["build_dir"]
+    return False
+
+
+def bool_menu(minifi_options: MinifiOptions) -> bool:
+    possible_values = [option_name for option_name in minifi_options.bool_options]
+    selected_values = [option.name for option in minifi_options.bool_options.values() if option.value == "ON"]
+    questions = [
+        inquirer.Checkbox(
+            "options",
+            message="MiNiFi C++ Options",
+            choices=possible_values,
+            default=selected_values
+        ),
+    ]
+
+    answers = inquirer.prompt(questions)
+    for bool_option in minifi_options.bool_options.values():
+        if bool_option.name in answers["options"]:
+            bool_option.value = "ON"
+        else:
+            bool_option.value = "OFF"
+
+    return False
```

Comment:
This menu needs guidance on how to exit from it after selecting the options.
![image](https://github.com/apache/nifi-minifi-cpp/assets/1170582/041bf975-0bcd-4b95-9dfd-b46f2915136d)

Reply:
I've added some clarifications to the title of this menu.
https://github.com/apache/nifi-minifi-cpp/pull/1681/files#diff-540b6c7fb2ac6e2c1e8c04096674ab473e4e16eaf2f06cff971b097ae65bc2cdR104